src/geom/SphPoint.cc

```diff
@@ -0,0 +1,244 @@
+            throw pexExcept::OutOfRangeError("Angle " + to_string(latitude.asDegrees()) +
+                " is not a valid latitude.");
+        }
+    }
+
+    SphPoint::SphPoint(Point3D const & vector) {
+        double const norm = sqrt(vector.distanceSquared(Point3D()));
+        if (!isfinite(norm)) {
+            stringstream buffer;
+            buffer << "NaN detected in vector " << vector << ".";
+            throw pexExcept::InvalidParameterError(buffer.str());
+        }
+        if (norm <= 0.0) {
+            stringstream buffer;
+            buffer << "Vector " << vector << " has no direction.";
+            throw pexExcept::InvalidParameterError(buffer.str());
+        }
+
+        double const x = vector.getX() / norm;
+        double const y = vector.getY() / norm;
+        double const z = vector.getZ() / norm;
+
+        // Need roundabout approach to handle _longitude = -1e-16
+        _longitude = wrap(atan2(y, x) * radians).asRadians();
+        _latitude = asin(z);
+    }
+
+    Point3D SphPoint::getVector() const noexcept {
+        return Point3D(cos(_longitude)*cos(_latitude),
+                       sin(_longitude)*cos(_latitude),
+                                       sin(_latitude));
+    }
+
+    Angle SphPoint::operator[](size_t index) const {
+        switch (index) {
+          case 0:
+            return getLongitude();
+          case 1:
+            return getLatitude();
+          default:
+            throw pexExcept::OutOfRangeError("Index " + to_string(index) + " must be 0 or 1.");
+        }
+    }
+
+    bool SphPoint::operator==(SphPoint const & other) const noexcept {
+        if (abs(_latitude) < HALFPI) {
+            return _longitude == other._longitude && _latitude == other._latitude;
+        } else {
+            return _latitude == other._latitude;
+        }
+    }
+
+    bool SphPoint::operator!=(SphPoint const & other) const noexcept {
+        return !(*this == other);
+    }
+
+    Angle SphPoint::bearingTo(SphPoint const & other) const {
+        if (abs(_latitude) >= HALFPI) {
+            stringstream buffer;
+            buffer << "Cannot calculate offset from pole " << *this << ".";
+            throw pexExcept::DomainError(buffer.str());
+        }
+
+        Angle const deltaLon = other.getLongitude()- this->getLongitude();
+
+        double const sinDelta1 = sin(      getLatitude().asRadians());
+        double const sinDelta2 = sin(other.getLatitude().asRadians());
+        double const cosDelta1 = cos(      getLatitude().asRadians());
+        double const cosDelta2 = cos(other.getLatitude().asRadians());
+
+        // Adapted from http://www.movable-type.co.uk/scripts/latlong.html
+        double const y = sin(deltaLon)*cosDelta2;
+        double const x = cosDelta1*sinDelta2 - sinDelta1*cosDelta2*cos(deltaLon);
+        return wrap(90.0*degrees - atan2(y, x)*radians);
+    }
+
+    Angle SphPoint::separationTo(SphPoint const & other) const noexcept {
```

Comment:
Is the "To" informative here? This is a symmetrical operation. I gently suggest `separation` or `angularSeparation`.